sdk/packages/selector-polkadot/src/adapter.ts

```diff
@@ -0,0 +1,490 @@
+          list.map((wallet) => ({
+            name: wallet.name,
+            icon: wallet.image.default,
+            deeplink: wallet.mobile,
+            link: wallet.homepage
+          }))
+        )
+        .catch(() => [] as MetadataWallet[])
+    ])
+
+    adapter._app = app
+    adapter._metadataWallets = metadataWallets
+    const recentWalletName = getRecentStandardWalletForNetwork(POLKADOT_NETWORK) ?? undefined
+
+    adapter._walletsList = metadataWallets.map((wallet) => ({
+      name: wallet.name,
+      icon: wallet.icon,
+      link: wallet.link,
+      deeplink: wallet.deeplink,
+      recent: recentWalletName === wallet.name
+    }))
+    const detectedWallets = getPolkadotWallets()
+    console.log({ detectedWallets })
+    adapter._injectedWallets = detectedWallets
+    // TODO do the deduplication
+    adapter._walletsList = adapter._walletsList.concat(
+      ...detectedWallets.map((wallet) => ({
+        name: wallet.name ?? 'Unknown',
+        icon: wallet.icon ?? '',
+        deeplink: null,
+        link: '',
+        recent: recentWalletName === wallet.name,
+        detected: true
+      }))
+    )
+
+    adapter._modal = new NightlyConnectSelectorModal(
+      adapter._walletsList,
+      appInitData.url ?? 'https://nc2.nightly.app',
+      {
+        network: QueryNetwork.POLKADOT,
+        name: POLKADOT_NETWORK,
+        icon: 'https://assets.coingecko.com/coins/images/4128/small/solana.png' // TODO add polka icon
+      },
+      anchorRef
+    )
+
+    return adapter
+  }
+
+  public static buildLazy = (
+    appInitData: AppInitData,
+    eagerConnectForStandardWallets?: boolean,
+    anchorRef?: HTMLElement | null
+  ) => {
+    const adapter = new NightlyConnectAdapter(appInitData, eagerConnectForStandardWallets)
+
+    adapter._loading = true
+
+    Promise.all([
+      AppPolkadot.build(appInitData),
+      AppPolkadot.getWalletsMetadata('https://nc2.nightly.app/get_wallets_metadata')
+        .then((list) =>
+          list.map((wallet) => ({
+            name: wallet.name,
+            icon: wallet.image.default,
+            deeplink: wallet.mobile,
+            link: wallet.homepage
+          }))
+        )
+        .catch(() => [] as MetadataWallet[])
+    ]).then(([app, metadataWallets]) => {
+      adapter._app = app
+      adapter._metadataWallets = metadataWallets
+      const recentWalletName = getRecentStandardWalletForNetwork(POLKADOT_NETWORK) ?? undefined
+
+      adapter._walletsList = metadataWallets.map((wallet) => ({
+        name: wallet.name,
+        icon: wallet.icon,
+        link: wallet.link,
+        deeplink: wallet.deeplink,
+        recent: recentWalletName === wallet.name
+      }))
+      console.log(adapter._walletsList)
+      const detectedWallets = getPolkadotWallets()
+      console.log({ detectedWallets })
+      adapter._injectedWallets = detectedWallets
+      // TODO do the deduplication
+
+      adapter._walletsList = adapter._walletsList.concat(
```

Comment:
Concatenation of these two lists is a wrong idea, because there willl be duplicates; it should be done like in function getWalletsList from selector-base/src/detection.ts

Reply:
done